include/mutex_yield.h

```diff
@@ -0,0 +1,13 @@
+#ifndef __MUTEX_YIELD_H__
+#define __MUTEX_YIELD_H__
+
+#include <common.h>
+#include <mips.h>
+
+typedef volatile uintptr_t mtx_yield_t;
```

Comment:
As I mentioned in the comments to the other commit this `volatile` shouldn't be needed.